src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,153 @@
+package duke.command;
+
+import duke.CommandType;
+import duke.DukeException;
+import duke.StringParser;
+import duke.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.time.DateTimeException;
+
+/**
+ * Command type add.
+ */
+public class AddCommand extends Command {
+
+    private final String command;
+    private final CommandType cmdType;
+
+    /**
+     * Add command builder.
+     *
+     * @param command Input string.
+     * @param cmdType Determine whether this command is TODO, EVENT or DEADLINE.
+     */
+    public AddCommand(String command, CommandType cmdType) {
+        this.command = command;
+        this.cmdType = cmdType;
+    }
+
+    /**
+     * Check the validity of this command and build a todo task.
+     *
+     * @return Todo task.
+     * @throws DukeException When command argument is invalid.
+     */
+    private Task convertToTodo() throws DukeException {
+        if (command.length() <= 5) {
+            throw new DukeException("Invalid argument: Argument field cannot be empty.");
+        } else {
+            String subStr = command.substring(5);
+            if (StringParser.isBlank(subStr)) {
+                throw new DukeException("Invalid argument: Content field is blank");
+            } else {
+                return new Todo(command.substring(5));
+            }
+        }
+    }
+
+    /**
+     * Check the validity of this command and build a event task.
+     *
+     * @return Event task.
+     * @throws DukeException When command argument is invalid.
+     */
+    private Task convertToEvent() throws DukeException {
+        if (command.length() <= 9) {
+            throw new DukeException("Invalid argument: Argument field cannot be empty.");
+        } else {
+            int indexOfAt = command.toLowerCase().indexOf("/at");
+            String subStrContent = command.substring(6, indexOfAt - 1);
+            String subStrTime = command.substring(indexOfAt + 4);
+            if (StringParser.isBlank(subStrContent)) {
+                throw new DukeException("Void argument: Content field is blank");
+            } else if (StringParser.isBlank(subStrTime)) {
+                throw new DukeException("Void argument: Time field is blank");
+            } else {
+                try {
+                    return new Event(subStrContent, StringParser.parseTime(subStrTime));
+                } catch (DateTimeException e) {
+                    throw new DukeException("Incorrect time format: Correct format is yyyy-MM-dd HHmm");
+                }
+            }
+        }
+    }
+
+    /**
+     * Check the validity of this command and build a deadline task.
```

Comment:
Not sure if the verb form here is not appropriate, similar problems encountered with other comments

Reply:
I think it should be "checks"? Good finding.